conpr/08_LE_Tasks/src/le/fj/Sum.java

```diff
@@ -26,27 +30,34 @@ protected Integer compute() {
             return sum; 
         } else {
             int mid = (hi + lo) / 2;
-            Sum firstHalf = new Sum(array, lo, mid);
-            firstHalf.fork();
-            Sum secondHalf = new Sum(array, mid, hi);
-            return secondHalf.compute() + firstHalf.join();
+            Sum firstHalf = new Sum(array, lo, mid, exec);
+            Sum secondHalf = new Sum(array, mid, hi, exec);
+            try {
+                return exec.submit(secondHalf).get() + exec.submit(firstHalf).get();
+            } catch (Exception e) {
+                e.printStackTrace();
+                return 0;
+            }
```

Comment:
Previously, when `firstHalf.fork()` and `secondHalf.compute()` were called, these tasks ran in the same ForkJoinPool as their parent Sum task. I confirmed this locally by adding printouts of [ForkJoinTask.getPool()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/ForkJoinTask.html#getPool()) [here](https://github.com/ponder-lab/FHNW-Java-Projekte/pull/1/files#diff-62aa5cd46958d4cbbc198e41583fdc44ce9c7e33dfd7b9505e0fa52a951fc612L30), [here](https://github.com/ponder-lab/FHNW-Java-Projekte/pull/1/files#diff-62aa5cd46958d4cbbc198e41583fdc44ce9c7e33dfd7b9505e0fa52a951fc612L32), and [here](https://github.com/ponder-lab/FHNW-Java-Projekte/pull/1/files#diff-62aa5cd46958d4cbbc198e41583fdc44ce9c7e33dfd7b9505e0fa52a951fc612R57).
To ensure that when we refactor, all subtasks of the original Sum task all run in the same ExecutorService, I add it as a parameter/property of the class, and chain ExecutorService.submit() and Future.get() like in the main method to get the result of each subtask.